packages/shared/src/components/post/write/CreatePostButton.tsx

```diff
@@ -38,7 +40,7 @@ export function CreatePostButton({
     verifyPermission(item, SourcePermissions.Post),
   );
 
-  if (!user || !squads?.length) {
+  if (!footer && (!user || !squads?.length)) {
```

Comment:
So footer is mantadory?
A bit unsure why this return is needed

Reply:
This check ensures the button is rendered if the footer prop is present. Otherwise, the following checks would kick in and the btn would not be rendered at all in the case where there is no user. 

Essentially we want to make sure that if this button is rendered in the footer, we don't care if there is no user, so it's always rendered